sw/applications/example_spi_obi_slave/main.c

```diff
@@ -0,0 +1,208 @@
+#define WRITE_SPI_SLAVE_REG_3 0x30
+#define READ_SPI_SLAVE_CMD 0xB
+
+
+typedef enum {
+    // Everithing went well
+    SPI_SLAVE_FLAG_OK                 = 0x0000,
+    // The SPI variabled passed was a null pointer
+    SPI_SLAVE_FLAG_NULL_PTR           = 0x0001,
+    //The SPI was not properly initalized
+    SPI_SLAVE_FLAG_NOT_INIT           = 0x0002,
+    //The target address is invalid
+    SPI_SLAVE_FLAG_ADDRESS_INVALID    = 0x0003,
+    // The CSID was out of the bounds specified inSPI_HOST_PARAM_NUM_C_S 
+    SPI_SLAVE_FLAG_CSID_INVALID       = 0x0004,
+    // The CMD FIFO is currently full so couldn't write command
+    SPI_SLAVE_FLAG_COMMAND_FULL       = 0x0008,
+    // The specified speed is not valid so couldn't write command
+    SPI_SLAVE_FLAG_SPEED_INVALID      = 0x0010,
+    // The TX Queue is full, thus could not write to TX register
+    SPI_SLAVE_FLAG_TX_QUEUE_FULL      = 0x0020,
+    // The RX Queue is empty, thus could not read from RX register
+    SPI_SLAVE_FLAG_RX_QUEUE_EMPTY     = 0x0040,
+    // The SPI is not ready
+    SPI_SLAVE_FLAG_NOT_READY          = 0x0080,
+    // The event to enable is not a valid event
+    SPI_SLAVE_FLAG_EVENT_INVALID      = 0x0100,
+    // The error irq to enable is not a valid error irq
+    SPI_SLAVE_FLAG_ERROR_INVALID      = 0x0200 
+} spi_slave_flags_e;
+
+spi_host_t* spi; //Removed specific memory assignement
+
+
+spi_slave_flags_e spi_slave_init(spi_host_t* spi_host);
+void spi_host_write();
+spi_slave_flags_e spi_slave_read(uint32_t addr, void* data, uint16_t length);
+bool check_address_validity(uint32_t addr, void* data, uint16_t length);
+
+
+int main(int argc, char *argv[])
+{   
+    spi_return_flags_e flags;
+    flags = spi_slave_init(spi);
+    if (flags != SPI_FLAG_OK){
+        printf("Failure to initialize\n Error code: %s", flags);
+        return EXIT_FAILURE;
+    }
+
+
+
+
+    return EXIT_SUCCESS;
+}
+
+void spi_host_write(){
+};
+
+spi_slave_flags_e spi_slave_read(uint32_t addr, void* data, uint16_t length){
+    // ??? Is this necessary? What happens when address is invalid or data exceeds memory? What is the size of the memory?
+    if (check_address_validity(addr, data, length) != true) return SPI_SLAVE_FLAG_ADDRESS_INVALID;
+
+
+    uint32_t wrap_length_cmd    = (WRITE_SPI_SLAVE_REG_2)                   //Write register 2 
+                                + ((LOWER_BYTE_16_BITS(length)) << 8)       //Wraplength low 
+                                + (WRITE_SPI_SLAVE_REG_3 << 16)             //Write register 3
+                                + (UPPER_BYTE_16_BITS(length) << 24);       //Wraplength high
+
+    // Load command to TX FIFO
+    spi_write_word(spi, wrap_length_cmd);
+    spi_wait_for_ready(spi);
+
+    // Set up segment parameters -> send commands
+    const uint32_t send_cmd_byte = spi_create_command((spi_command_t){
+        .len        = 1,                    // 1 Byte (The SPI SLAVE IP can only take one CMD byte at a time)
```

Comment:
I think that if you want to write N bytes you have to pass to the function N-1, look at the examples